lib/posix/timer.c

```diff
@@ -3,11 +3,31 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
+
+#ifdef CONFIG_PICOLIBC
+/* FIXME: Please see Issue #46910 */
+#define _POSIX_MONOTONIC_CLOCK
+#endif
+
 #include <zephyr/kernel.h>
 #include <errno.h>
+#include <signal.h>
 #include <string.h>
 #include <zephyr/sys/printk.h>
-#include <zephyr/posix/time.h>
+#include <time.h>
```

Comment:
See above -- if you add the right #defines before #including time.h, you should get the functions you need defined.